functions/src/cron.ts

```diff
@@ -39,18 +39,71 @@ export class CronHandler {
     this.config = config;
   }
 
-  async handleStaleIssues(org: string, name: string): Promise<types.Action[]> {
+  async processIssues(org: string, name: string) {
     log.debug(`Processing issues for ${org}/${name}`);
 
     // Get the configuration for this repo
     const cleanupConfig = this.config.getRepoCleanupConfig(org, name);
     if (!cleanupConfig || !cleanupConfig.issue) {
-      log.debug(`No stale issues config for ${org}/${name}`);
+      log.debug(`No issue cleanup config for ${org}/${name}`);
       return [];
     }
     const issueConfig = cleanupConfig.issue;
 
-    // Aggregate all the actions we need to perform
+    const lockActions = await this.handleClosedIssues(org, name, issueConfig);
+    const staleActions = await this.handleStaleIssues(org, name, issueConfig);
+
+    return [...lockActions, ...staleActions];
+  }
+
+  async handleClosedIssues(
+    org: string,
+    name: string,
+    issueConfig: types.IssueCleanupConfig
+  ): Promise<types.Action[]> {
+    if (!issueConfig.lock_days) {
+      log.debug(`No issue locking config for ${org}/${name}`);
+      return [];
+    }
+
+    const lockDays = issueConfig.lock_days;
+    const lockMillis = lockDays * 24 * 60 * 60 * 1000;
+
+    const actions: types.Action[] = [];
+
+    const issues = await this.gh_client.getIssuesForRepo(org, name, "closed");
+    for (const issue of issues) {
+      if (issue.closed_at === null) {
+        log.warn(
+          `Closed issue ${org}/${name}/${issue.number} has no closed_at.`
+        );
+        continue;
+      }
+
+      const closedAtStr = "" + issue.closed_at;
+      const closedAtMs = new Date(closedAtStr).getTime();
+      const nowMs = new Date().getTime();
+
+      if (nowMs - closedAtMs > lockMillis) {
+        actions.push(
+          new types.GithubLockAction(
+            org,
+            name,
+            issue.number,
+            `Issue was closed at ${closedAtStr} which is more than ${lockDays} ago`
```

Comment:
nit: what is the purpose of this text? I think it's not used yet. For us it's fine to just lock after lockDays. We currently do not add a comment to tell people why it has been locked. I guess after lockDays that's fine anyway.

Reply:
This is the "reason" which is only displayed in the audit log, not shown on GitHub.